python/MooseDocs/testing.py

```diff
@@ -37,11 +37,28 @@ def setUpClass(cls):
     os.chdir(os.path.join(MooseDocs.MOOSE_DIR, 'docs'))
 
     config = MooseDocs.yaml_load('moosedocs.yml')
+    cls.updateConfig(config)
 
     extensions, extension_configs = MooseDocs.get_markdown_extensions(config)
     cls.parser = markdown.Markdown(extensions=extensions, extension_configs=extension_configs)
     os.chdir(cwd)
 
+  @classmethod
+  def updateConfig(cls, config):
```

Comment:
This can be simplified if we take this concept and move it down a layer. On 42 above, the extensions and associated configurations are pulled apart from the main configuration, so we can use a config update function on this instead.

```
@classmethod
def updateExtensionConfigs(extension_config):
  if "MooseDocs.extensions.MooseMarkdown" in extension_config:
     extension_config['MooeDocs...']['macro_files'] = ...
```